### PR TITLE
Update pycord to 2.5.0, also allowing python 3.12 to be used

### PR DESCRIPTION
**Pull request by ToothyDev**
Just a little dependency maintenance to get it more up to date

### CODE REVIEW

**Comment by Soapy7261** [comment]
i should remove codeql...

**Comment by Soapy7261** [comment]
wait it is removed wtf?

**Comment by ToothyDev** [comment]
I accepted the license, twice, and he still says nuh-uh

**Comment by JoshuaSlui** [comment]
Definitely should, but we're holding out on this pull request until monday.
It's not really high priority right now due to the circumstances.

**Review by Soapy7261 — DISMISSED** [review]

File: cogs/blacklist.py [hunk 1]
```diff
@@ -2,16 +2,17 @@
 from discord import slash_command, option
 from discord.ext import commands
 from discord.ext.bridge import Bot
+
```
Comment (JoshuaSlui) [inline cogs/blacklist.py:5]: Why?

File: cogs/blacklist.py [hunk 2]
```diff
@@ -2,16 +2,17 @@
...
 
 
-class Blacklist(commands.Cog):
+class Blacklist(discord.Cog):
     def __init__(self, bot: Bot):
         self.bot = bot
 
     @slash_command()
     @commands.is_owner()
-    @option("server", discord.Guild, description="The server to blacklist")
```
Comment (JoshuaSlui) [inline cogs/blacklist.py]: This was consciously chosen so we could just input the guild name. That actually works - as i just confirmed. Please dont change this.

File: cogs/checkip.py [hunk 3]
```diff
@@ -1,12 +1,14 @@
 from asyncio import wait_for
+
+import discord
 from discord import slash_command, option, Embed
-from discord.ext import commands
 from discord.ext.commands import Bot
+
 from utilities.data import Colors, get_server_status
 from utilities.utility import check_ip
```
Comment (JoshuaSlui) [inline cogs/checkip.py:8]: Please group the respective imports together. Why the weird newlines between imports?

File: cogs/cogs.py [hunk 4]
```diff
@@ -1,12 +1,14 @@
 from os import listdir
-from discord.ext.commands import slash_command
-from discord.ext import commands
+
+import discord
 from discord import Option
 from discord.ext.bridge import Bot
+from discord.ext.commands import slash_command
+
 from utilities.data import get_data
```
Comment (JoshuaSlui) [inline cogs/cogs.py:8]: Again, group the remaining imports together

File: cogs/info.py [hunk 5]
```diff
@@ -1,11 +1,13 @@
 from asyncio import wait_for
+
+import discord
 from discord import slash_command, Embed
-from discord.ext import commands
 from discord.ext.commands import Bot
+
 from utilities.data import Colors, get_server_status
```
Comment (JoshuaSlui) [inline cogs/info.py:7]: You know the deal

File: cogs/status.py [hunk 6]
```diff
@@ -1,13 +1,15 @@
 from asyncio import wait_for
+
+import discord
 from discord import slash_command, option, Embed, utils as dutils
-from discord.ext import commands
 from discord.ext.commands import Bot
+
 from utilities.data import remove_colors_from_string, Colors, get_server_status
 from utilities.database import selector
 from utilities.utility import check_ip
```
Comment (JoshuaSlui) [inline cogs/status.py:9]: Here as well.

File: requirements.txt [hunk 7]
```diff
@@ -1,4 +1,4 @@
-py-cord[speed]==2.4.1
+py-cord==2.5.0
 python-dotenv==0.21.1
-mcstatus==11.0.1
-mysql-connector-python==8.0.32
+mcstatus==11.1.1
+mysql-connector-python==8.3.0
```
Comment (JoshuaSlui) [inline requirements.txt]: Use >= for these

**Review by JoshuaSlui — CHANGES_REQUESTED** [review]

File: cogs/blacklist.py (hunk 1 above)
Comment (ToothyDev) [reply cogs/blacklist.py:5]: To seperate third party imports (discord) from local imports (utilities.database)

File: cogs/checkip.py (hunk 3 above)
Comment (ToothyDev) [reply cogs/checkip.py:8]: To seperate third party imports (discord) from local imports (utilities.data) and built-in imports (asyncio)

File: cogs/cogs.py (hunk 4 above)
Comment (ToothyDev) [reply cogs/cogs.py:8]: Done to seperate third party imports (discord) from local imports (utilities.data) and built-in imports (asyncio)

File: cogs/info.py (hunk 5 above)
Comment (ToothyDev) [reply cogs/info.py:7]: Done to seperate third party imports (discord) from local imports (utilities.data) and built-in imports (asyncio)

File: cogs/status.py (hunk 6 above)
Comment (ToothyDev) [reply cogs/status.py:9]: Done to seperate third party imports (discord) from local imports (utilities.data) and built-in imports (asyncio)

File: requirements.txt (hunk 7 above)
Comment (ToothyDev) [reply requirements.txt]: Not ~= for compatibility equality?

Comment (ToothyDev) [reply requirements.txt]: ~= will allow any greater minor version, but no major version update. That way updates never break existing code ideally. >= allows e.g. pycord 3.0 to be used too, which has breaking changes as a major update

Comment (JoshuaSlui) [reply requirements.txt]: No.

Comment (JoshuaSlui) [reply requirements.txt]: I'm aware that breaking changes _could_ break it, but principly, when a package is updated to that point it should be updated here as well anyways. This'll both keep us on the ropes for that and make sure we're up to date with anything, even if major releases don't necessarily have breaking changes that can interfere with our code - which also happens often enough.

**Review by JoshuaSlui — DISMISSED** [review]

**Comment by JoshuaSlui** [comment]
@Soapy7261 You see anything that should need changing here?

**Comment by Soapy7261** [comment]
uhhhhm

**Comment by Soapy7261** [comment]
will take a look in a bit

**Comment by JoshuaSlui** [comment]
I think you and I have mildly different definitions of 'a bit' 🤔

**Review by JoshuaSlui — APPROVED** [review]

**Comment by JoshuaSlui** [comment]
I'll merge this so we can speed up the remaining pull requests.
There is no significant change in this pull request that warrants any further delay.

**Pull request merged**